internal/iso/iso.go

```diff
@@ -122,60 +121,84 @@ func (h *Handler) RoundTrip(req *http.Request) (*http.Response, error) {
 	// RoundTripper needs a Transport to execute a HTTP transaction
 	// For our use case the default transport will suffice.
 	resp, err := http.DefaultTransport.RoundTrip(req)
-	// resp, err := h.RoundTripper.RoundTrip(req)
 	if err != nil {
-		h.Logger.Info("HTTP request didn't receive a response", "sourceIso", h.SourceISO, "error", err)
+		log.Info("issue getting the source ISO", "sourceIso", h.SourceISO, "error", err)
 		return nil, err
 	}
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusPartialContent {
+		log.Info("the request to get the source ISO returned a status other than ok (200) or partial content (206)", "sourceIso", h.SourceISO, "status", resp.Status)
+		return resp, nil
+	}
+	// by setting this header we are telling the logging middleware to not log its default log message.
+	// we do this because there are a lot of partial content requests.
+	resp.Header.Set("X-Global-Logging", "false")
 
 	if req.Method == http.MethodHead {
-		// Fuse client typically make a HEAD request before they start requesting content.
-		h.Logger.V(1).Info("HTTP HEAD request received, patching only occurs on 206 requests")
+		// Fuse clients typically make a HEAD request before they start requesting content.
+		log.Info("HTTP HEAD method received")
 		return resp, nil
 	}
 
-	// roundtripper should only return error when no response from the server
-	// for any other case just log the error and return 404 response
-	if resp.StatusCode == http.StatusPartialContent {
-		b, err := io.ReadAll(resp.Body)
-		if err != nil {
-			h.Logger.Error(err, "reading response bytes", "response", resp.Body)
-			return &http.Response{
-				Status:     fmt.Sprintf("%d %s", http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError)),
-				StatusCode: http.StatusInternalServerError,
-				Body:       http.NoBody,
-				Request:    req,
-			}, nil
-		}
-		if err := resp.Body.Close(); err != nil {
-			h.Logger.Error(err, "closing response body", "response", resp.Body)
-			return &http.Response{
-				Status:     fmt.Sprintf("%d %s", http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError)),
-				StatusCode: http.StatusInternalServerError,
-				Body:       http.NoBody,
-				Request:    req,
-			}, nil
-		}
+	// 0.002% of the time we log a 206 request message.
+	// In testing, it was observed that about 3000 HTTP 206 requests are made per ISO mount.
+	// 0.002% gives us about 5, +/- 3, log messages per ISO mount.
+	// We're optimizing for showing "enough" log messages so that progress can be observed.
+	if p, _ := randomPercentage(10000); p < 0.002 {
```

Comment:
An additional code comment on why swallowing this error is safe and insignificant could be useful.